src/lib/confmgt/unitparse.c

```diff
@@ -145,7 +146,14 @@ config_parse_units(const char *val, const unit_table_t *u, int *ok)
       if (use_float)
         v = (uint64_t)(u->multiplier * d);
       else
-        v *= u->multiplier;
+        v = tor_mul_u64_nowrap(v, u->multiplier);
+
+      if ( v > SSIZE_MAX ) {
```

Comment:
Would it be okay to check for `v > UINT64_MAX` instead of `SSIZE_MAX` ?

[Trac comment](https://trac.torproject.org/projects/tor/ticket/30920#comment:13) regarding using SSIZE_MAX instead of UINT64_MAX. Probably the difference between SSIZE_MAX in 64-bit machines and 32-bit machines are causing some tests to fail.